src/03-ticket-calculator.js

```diff
@@ -54,7 +55,134 @@ const exampleTicketData = require("../data/tickets");
     calculateTicketPrice(tickets, ticketInfo);
     //> "Entrant type 'kid' cannot be found."
  */
-function calculateTicketPrice(ticketData, ticketInfo) {}
+function calculateTicketPrice(ticketData, ticketInfo) {
+//   let result = 0;
+  
+//   for (let i = 0; i < ticketInfo.length; i++){
+
+//     let totalPrice = 0;
+
+//     for (let j = 0; j < ticketInfo[i].extras.length; j++){
+      
+//       let extraPrice = 0;
+
+//       if (ticketData.extras.hasOwnProperty(ticketInfo[i].extras[j])){
+
+//         extraPrice += Number(ticketData.extras[ticketInfo[i].extras[j]])
+
+//         } else {
+
+//         return `Extra type '${ticketInfo[i].extras[j]}' cannot be found.`
+
+//       }
+
+//     totalPrice += extraPrice
+  
+//     }
+
+//   if(ticketData.hasOwnProperty(ticketInfo[i].ticketType) && ticketData[ticketInfo[i].ticketType].priceInCents.hasOwnProperty(ticketInfo[i].entrantType)){
+
+//     totalPrice += Number(ticketData[ticketInfo[i].ticketType].priceInCents[ticketInfo[i].entrantType])
+
+//     } else if (!ticketData.hasOwnProperty(ticketInfo[i].ticketType)){
+
+//       return `Ticket type '${ticketInfo[i].ticketType}' cannot be found.`
+
+//     } else if (!ticketData[ticketInfo[i].ticketType].priceInCents.hasOwnProperty(ticketInfo[i].entrantType )){
+
+//       return `Entrant type '${ticketInfo[i].entrantType}' cannot be found.`
+
+//     }
+
+//     result += totalPrice
+
+//   }
+
+//   return result
+
+// }
+  if (!Object.keys(ticketData).includes(ticketInfo.ticketType)){
+    return `Ticket type '${ticketInfo.ticketType}' cannot be found.`
```

Comment:
another way to check if the property exist and therefore is an appropriate ticketType is by checking if `ticketData[ticketInfo.ticketType` and if it falsey or `undefined` then it is safe to say that the ticketType is incorrect.